Redo "badge app 101" instructions (#324)
emfcamp/badge-2024-documentation: docs/tildagon-apps/simulate.md

title: Simulate your app
---

You can run badge apps on your computer with the local simulator from the [badge-2024-software](https://github.com/emfcamp/badge-2024-software/tree/main/sim) repo.

> There is also a [web emulator](https://emulator.badge.emfcamp.org/) but as of 21st June 2026 it does not support running your own app.

Reviewer: ```suggestion
There is also a [web emulator](https://emulator.badge.emfcamp.org/).
Use it to explore the badge UI and try apps from the [app store](https://apps.badge.emfcamp.org/) before installing them on your badge. Once you [publish your apps](./publish.md) you can also try them there.
```

### Installation

1. Install [Python](https://www.python.org/downloads/).

> 1. Install pipenv if you didn't already: `pip install --user pipenv`.

Reviewer: ```suggestion
1. Install pipenv: `pip install --user pipenv`.
```

> 1. You should now be able to run `pipenv --version` in a terminal and it shouldn't error. If you don't, try `python -m pipenv --version`. If that works but `pipenv --version` didn't, you'll need to remember to type `python -m` before `pipenv` for the rest of this guide.

Reviewer: ```suggestion
1. Run `pipenv --version` in a terminal. If you don't see a version number, try `python -m pipenv --version`. If that works but `pipenv --version` didn't, you'll need to remember to type `python -m` before `pipenv` for the rest of this guide.
```
